hatchet_sdk/clients/workflow_listener.py

```diff
@@ -123,6 +122,9 @@ async def _init_producer(self):
 
                             workflow_event: WorkflowRunEvent = t.result()
 
+                            if workflow_event is cygrpc.EOF:
```

Comment:
We weren't casing on `EOF` errors, which could cause exceptions to be raised out of the listener.